### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Ashishgupta08**
Live URL - [Quiz App](https://quizo-app.netlify.app/)

Test Credentials:
username - admin
password: admin

Do provide your valuable feedback to improve the app.
ThankYou!

### CODE REVIEW

File: models/leaderboard.model.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const mongoose = require('mongoose')
+const { Schema } = mongoose
```
Comment (adarshwritescode) [inline models/leaderboard.model.js:2]: `const { Schema , model } = require('mongoose') `

File: routes/leaderboard.router.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+router.post('/', authorizedUser, async (req, res) => {
+  const { quizId, leaderBoard } = req.body;
+  try {
+    const quizLeaderBoard = await LeaderBoard.findOne({ quizId: quizId })
+    if (quizLeaderBoard === null) {
+      const newLeaderBoard = new LeaderBoard({
+        quizId: quizId,
+        leaderBoard: leaderBoard
+      })
+      const data = await newLeaderBoard.save();
```
Comment (adarshwritescode) [inline routes/leaderboard.router.js:32]: only `await newLeaderBoard.save()` will work , we're not using 'data' anyway

File: routes/user.router.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+const bcrypt = require("bcrypt");
+const router = express.Router();
+const { User } = require('../models/user.model')
+const { authorizedUser } = require('../utils/authorizedUser')
+const secret = process.env['SECRET'];
+
+router.post('/login', async (req, res) => {
+  const { username, password } = req.body
+  try {
+    const user = await User.findOne({ username: username })
```
Comment (adarshwritescode) [inline routes/user.router.js:12]: you can simply pass `const user = await User.findOne({ username })` no need to pass username again (ES6)

File: routes/user.router.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    const salt = await bcrypt.genSalt(10);
+    const encryptedPassword = await bcrypt.hash(password, salt);
+    const newUser = new User({
+      username: username,
+      password: encryptedPassword,
+      name: name,
+      email: email,
+      contact: contact,
+      gender: gender
+    })
```
Comment (adarshwritescode) [inline routes/user.router.js:79]: `const newUser = new User({username,password:encryptedPassword,name,email,contact,gender})`  (ES6 😁)

**Review by WhoAdarshPandya — COMMENTED** [review]
nicely done Ashish! suggesting few changes here

- signup not working (gives 409)
- validate email on signup / enforce user for alphanum password
- deploy on heroku instead of repl (repl is slow sometimes)

**Pull request merged**